gst/nnstreamer/tensor_query/tensor_query_client.c

```diff
@@ -488,6 +398,118 @@ _client_retry_connection (GstTensorQueryClient * self)
   return ret;
 }
 
+/**
+ * @brief Parse caps from received event data.
+ */
+static gchar *
+_nns_edge_parse_caps (gchar * caps_str, gboolean is_src)
+{
+  gchar **strv = g_strsplit (caps_str, "@", -1);
+  gint num = g_strv_length (strv), i;
+  gchar *find_key = NULL;
+  gchar *ret_str = NULL;
+
+  find_key =
+      is_src ==
+      TRUE ? g_strdup ("query_server_src_caps") :
+      g_strdup ("query_server_sink_caps");
+
+  for (i = 1; i < num; i += 2) {
+    if (0 == g_strcmp0 (find_key, strv[i])) {
+      ret_str = g_strdup (strv[i + 1]);
+      break;
+    }
+  }
+
+  g_free (find_key);
+  g_strfreev (strv);
+
+  return ret_str;
+}
+
+/**
+ * @brief nnstreamer-edge event callback.
+ */
+static int
+_nns_edge_event_cb (nns_edge_event_h event_h, void *user_data)
+{
+  nns_edge_event_e event_type;
+  int ret = 0;
+  gchar *caps_str = NULL;
+  GstTensorQueryClient *self = (GstTensorQueryClient *) user_data;
+
+  if (0 != nns_edge_event_get_type (event_h, &event_type)) {
+    nns_loge ("Failed to get event type!");
+    return -1;
```

Comment:
recommend to return nns-edge error code here

Reply:
Updated.